sound/soc/sof/core.c

```diff
@@ -215,16 +216,17 @@ int snd_sof_create_page_table(struct snd_sof_dev *sdev,
 		 */
 		u32 idx = (5 * i) >> 1;
 		u32 pfn = snd_sgbuf_get_addr(dmab, i * PAGE_SIZE) >> PAGE_SHIFT;
-		u32 *pg_table;
+		u8 *pg_table;
 
 		dev_vdbg(sdev->dev, "pfn i %i idx %d pfn %x\n", i, idx, pfn);
 
-		pg_table = (u32 *)(page_table + idx);
+		pg_table = (u8 *)(page_table + idx);
 
 		if (i & 1)
-			*pg_table |= (pfn << 4);
```

Comment:
I think adding one comment here saying how the pages are compressed would be very helpful. I remember that I spent some time to figure out why if i is odd then you need to shift by 4. And now I don't even remember if I figure that out.

Reply:
yeah, it's really not clear. It took me a while to figure out that 2.5 bytes for each compressed value means every other value needs to be shifted by 4 bits (half a byte) and ORed, and likewise that the offsets are 2,5,7,10, etc.
Someone wanted to be smart and only managed to confuse everyone else I guess...

Reply:
@dbaluta i can empathize with you on that. Everytime I read this bit of code, I need a piece of paper to work out whats going on

Reply:
oh, I thought it was a nice little exercise to keep the reader awake, but ok, if you want... ;-)

Reply:
Legacy physical page compression used on the HSW/BDW firmware, used to describe deep buffers in a small size. This has been inherited by SOF.